src/huggingface_hub/_commit_api.py

```diff
@@ -386,16 +386,40 @@ def _upload_lfs_object(operation: CommitOperationAdd, lfs_batch_action: dict, to
         return
     upload_action = lfs_batch_action["actions"].get("upload")
     verify_action = lfs_batch_action["actions"].get("verify")
-    with operation.as_file(with_tqdm=True) as fileobj:
-        logger.debug(f"Uploading {operation.path_in_repo} as LFS file...")
-        lfs_upload(
-            fileobj=fileobj,
-            upload_action=upload_action,
-            verify_action=verify_action,
-            upload_info=upload_info,
-            token=token,
-        )
-        logger.debug(f"{operation.path_in_repo}: Upload successful")
+    if HF_HUB_ENABLE_HF_TRANSFER:
+        if not isinstance(operation.path_or_fileobj, str):
+            raise ValueError("When using hf_transfer, you should not open the file beforehand and pass the file path as a string instead")
+        try:
+            # Upload file using an external Rust-based package. Upload is faster
+            # but support less features (no progress bars).
+            from hf_transfer import upload
+
+            logger.debug(f"Uploading {operation.path_in_repo} as LFS file...")
+            upload(operation.path_or_fileobj, upload_action, verify_action, upload_info.sha256, upload_info.size, token, max_files=128, parallel_failures=127, max_retries=5)
```

Comment:
>Did I understand it correctly?

Yup. `max_files` means max open file descriptors basically, which is the number of chunk uploads that will run concurrently. `parallel_failures` is what you described, the maximum number of errors that can happen at the same time. If you reach `parallel_failures` errors, we assume that there is an outage on the server you're trying to upload and therefore cancel the upload without retrying to avoid overloading an already unstable / dead system.

Reply:
If there are less than `parallel_failures` errors occurring, chunk uploads will be retries `max_retries` times.